go/tricorder/metric_test.go

```diff
@@ -1737,6 +1775,16 @@ func TestListNoChangeSubType(t *testing.T) {
 	t.Error("panic expected")
 }
 
+func TestNoAssignedUnitNoAdd(t *testing.T) {
+	bucketer := NewGeometricBucketer(1, 1000)
+	dist := newDistribution(bucketer, false)
+	defer func() {
+		recover()
```

Comment:
I think it's better to test the kind of panic, to ensure we got the expected panic.